cache_example_test.go

```diff
@@ -7,49 +7,48 @@ import (
 	"os"
 
 	"github.com/tetratelabs/wazero"
-	"github.com/tetratelabs/wazero/experimental"
 )
 
 // This is a basic example of using the file system compilation cache via WithCompilationCacheDirName.
 // The main goal is to show how it is configured.
-func Example_withCompilationCacheDirName() {
+func Example_withCache() {
 	// Prepare a cache directory.
 	cacheDir, err := os.MkdirTemp("", "example")
 	if err != nil {
 		log.Panicln(err)
 	}
 	defer os.RemoveAll(cacheDir)
 
-	// Append the directory into the context for configuration.
-	ctx, err := experimental.WithCompilationCacheDirName(context.Background(), cacheDir)
-	if err != nil {
-		log.Panicln(err)
+	ctx := context.Background()
+
+	// Creates a new cache context with the file cache directory configured.
+	cache := wazero.NewCache()
+	defer cache.Close(ctx)
+	if err := cache.WithCompilationCacheDirName(cacheDir); err != nil {
+		log.Fatal(err)
 	}
 
+	// Creates a runtime configuration to create multiple runtimes.
+	config := wazero.NewRuntimeConfig().
+		WithCache(cache)
```

Comment:
I'd like the cache to not actually do anything until the runtime is created, if possible. e.g. don't create the directory until you have a runtime. ack there's some tension here and it implies some internal types

Reply:
the problem here is that NewRuntime doesn't return error.. so wdyt on handling the error on directory creation (or not a dir error)?

Reply:
maybe it is better to make NewRuntime return an error. It just was we had no reason to before.

Then the logic is similar to instantiate which bundles up any sort of problem leading to it.

Reply:
If we make NewRuntime return an error we can add MustRuntime to allow people to assume it won't and panic. it is an api break anyway when we remove namespace, and I can help with the rounds on it.

Reply:
or.. we accept if the ony thing that can fail is directory creation, we add error return to the NewCompileCacheFromDisk and leave it at that.. is that the only possible error? If so mainly this turns more of the work to defer using it until NewRuntime.